master/buildbot/worker/docker.py

```diff
@@ -41,6 +41,7 @@
 except ImportError:
     docker = None
     client = None
+    docker_py_version = 0.0
 
 
 def _handle_stream_line(line):
```

Comment:
the test should not pass, it will be skipped.
That means buildbot +  docker will not be supported on gentoo.

I don't think this is what you want. The patch is good though, as docker should not be mandatory for running buildbot (and running the tests).

Reply:
I have added the docker dependency to the test deps, and made it optional for normal run time.
setup.py dopes not list the docker-py dependency though.  You probably should list it as an optional
dependency.

Reply:
right, we have a lot of optional dependencies, which we do not describe very well (its in the doc, in the relevent parts)